internal/signerverifier/ed25519/ed25519.go

```diff
@@ -0,0 +1,120 @@
+package ed25519
+
+import (
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/adityasaky/gittuf/internal/signerverifier"
+	"github.com/adityasaky/gittuf/internal/tuf"
+	"github.com/secure-systems-lab/go-securesystemslib/cjson"
+)
+
+type Ed25519SignerVerifier struct {
+	keyID   string
+	private ed25519.PrivateKey
+	public  ed25519.PublicKey
+}
+
+// NewSignerVerifierFromSecureSystemsLibFormat creates an Ed25519SignerVerifier
+// from a key serialized in the custom securesystemslib format.
+func NewSignerVerifierFromSecureSystemsLibFormat(keyContents []byte) (*Ed25519SignerVerifier, error) {
+	var k interface{}
+	if err := json.Unmarshal(keyContents, &k); err != nil {
+		return &Ed25519SignerVerifier{}, err
+	}
+
+	t := k.(map[string]interface{})
+
+	var keyID string
+	if _, ok := t["keyid"]; ok {
+		keyID = t["keyid"].(string)
```

Comment:
Instead of doing this type casting, I'd just define a struct and unmarshal there, then do the transformation into the SV.

Reply:
I've taken a pass at this. I must note that it's not clean because I didn't want to include private key support in the internal/tuf package.